### ISSUE TITLE
--p2p-host-ip does nto seem reflected in logs

### ISSUE THREAD

**Issue by tzapu** [prysmaticlabs/prysm#4267]
hi

i have used --p2p-host-ip to sepcify an external ip but the multi addr seems to still point at the docker internal ip 
`time="2019-12-12 11:14:31" level=info msg="Node started p2p server" multiAddr="/ip4/172.17.0.2/tcp/8546/p2p/16Uiu2HAm1oYE38s5JE9TkhoY9xGm9w6FH469hpJyXmpApaAeUSFX" prefix=p2p`
is this correct?
sync up seemed a lot faster, but might be just an impression

**Comment by nisdas** [comment]
hey so `p2p-host-ip` is added to the libp2p address factory and that is still used. Just that we do not log it over here. We probably should do that too.

**Comment by tzapu** [comment]
but is the multiaddr correct in this case where i manually specify an external public ip?

**Comment by nisdas** [comment]
So this multiaddress will not work in that case, since this is an internal IP, any external peer cannot reach you with this multiaddress

**Comment by tzapu** [comment]
then i would assume whatever's set in p2p-host-ip should propagate to this as well. the port i set did  (8546)

**Comment by nisdas** [comment]
Yeap it will propagate to external peers. So what happens is that when initially connecting with another peer, all the multiaddresses are exchanged between peers, including one with your internal docker ip and host ip.

**Comment by tzapu** [comment]
sorry, what i meant to say is that since i am starting the node with `-p2p-tcp-port PORT --p2p-host-ip IP.ADD.R.ESS those should be appearing in the multiaddr string as well, but only the port i set does, the ip i set does not.

**Comment by mcdee** [comment]
There are two addresses involved here.

One is the address to which the beacon node binds.  This will always be the address of one of the interfaces on your server, and can be confirmed by checking for the listening port:

```
$ egrep 'Node started' ~/beacon-chain.out 
{"level":"info","msg":"Node started p2p server","multiAddr":"/ip4/10.0.0.189/tcp/13000/p2p/16Uiu2HAmFLfXi8CFGhxBDUdd5MLcA27SizDyLqmyZAh5XBQ5gKMm","prefix":"p2p","time":"2019-12-12T11:40:10Z"}
$ lsof -ni :13000
COMMAND     PID USER   FD   TYPE   DEVICE SIZE/OFF NODE NAME
beacon-ch 29688  jgm    5u  IPv4 13993397      0t0  TCP 10.0.0.189:13000 (LISTEN)
```

The other is the address which is sent over the libp2p network and is used for other peers to connect to you.  This latter address is the one that is set by `--p2p-host-ip`

**Comment by tzapu** [comment]
ok, cool, thanks guys, makes sense now.
as @nisdas said it would be useful to log the external one as well then
feel free to close this

**Comment by nisdas** [comment]
I am going to leave this open, as we do want to log both of them, not just the ip that we bind to.

**Comment by rauljordan** [comment]
Resolved by #4553
